src/codegen/fast_math.hpp

```diff
@@ -86,25 +75,26 @@ static constexpr float PX6expf = 5.0000001201E-1f;
 
 static constexpr float LOG2EF = 1.44269504088896341f;  // 1/ln(2)
 
-static inline double egm1(double x, double px) {
-    x -= px * C1;
-    x -= px * C2;
+static inline double egm1(double x, double n) {
+    // this cannot be reordered for the cancellation trick to work
+    volatile double g = x - n * C1;
+    g -= n * C2;
```

Comment:
yeah exactly, if the compiler re-organize the ops it might remove the cancellation trick. It was happening with the `intel` compiler and `-02` or higher (that's why the CI was failing).

Reply:
@castigli : for the completeness, could you mention `cancellation trick` or briefly mention how Intel compiler was transforming the code?

Reply:
@pramodk , I added a short explanation, let me know if it is sufficient. Also, to be precise, I was using the wrong name for the trick! The splitting of coefficient seems to be called `double-double` in literature.

Reply:
thats very clear now. thanks!